platform/swift/source/reports/DiagnosticEventReporter.m

```diff
@@ -0,0 +1,295 @@
+  NSMutableSet <NSString *>* images = [NSMutableSet new];
+  NSArray *callStacks = dict_for_key(crash, @"callStackTree")[@"callStacks"];
+  for (NSDictionary *thread in callStacks) {
+    bool didCrash = [number_for_key(thread, @"threadAttributed") boolValue];
+    NSDictionary *frame = [array_for_key(thread, @"callStackRootFrames") firstObject];
+    uint64_t frameCount = count_frames(frame);
+    BDStackFrame *stack = (BDStackFrame *)calloc(frameCount, sizeof(BDStackFrame));
+    uint32_t index = 0;
+
+    while ([frame isKindOfClass:[NSDictionary class]]) {
+      NSString *binaryName = string_for_key(frame, @"binaryName");
+      NSString *binaryUUID = string_for_key(frame, @"binaryUUID");
+      NSNumber *address = number_for_key(frame, @"address");
+      NSNumber *offset = number_for_key(frame, @"offsetIntoBinaryTextSegment");
+      if (binaryName && binaryUUID && address && offset) {
+        if (![images containsObject:binaryUUID]) {
+          BDBinaryImage image = {
+            .id = cstring_from(binaryUUID),
+            .path = cstring_from(binaryName),
+            .load_address = [address unsignedLongLongValue] - [offset unsignedLongLongValue],
+          };
+          bdrw_add_binary_image(handle, &image);
+          [images addObject:binaryUUID];
+        }
+      } else {
+        break; // if the frame is invalid, it's time to leave
+      }
+      stack[index++] = (BDStackFrame) {
+        .image_id = cstring_from(binaryUUID),
+        .frame_address = [address unsignedLongLongValue],
+        .type_ = 2, // FrameType.DWARF
+      };
+      frame = [array_for_key(frame, @"subFrames") firstObject];
+    }
+    if (didCrash) {
+      bdrw_add_error(handle, cstring_from(name), cstring_from(reason), 0, index, stack);
+    } else {
+      BDThread thread = { .quality_of_service = -1 };
+      bdrw_add_thread(handle, [callStacks count], &thread, index, stack);
+    }
+    free(stack);
+  }
+}
+
+static int8_t architecture_constant(NSString *arch) {
+  if (!arch) {
+    return /* Architecture.Unknown */ 0;
+  }
+  return [arch containsString:@"arm64"] ? /* Architecture.arm64 */ 2 : /* Architecture.x86_64 */ 4;
+}
+
+static void serialize_device_metrics(BDProcessorHandle handle, NSDictionary *metadata, NSTimeInterval timestamp) {
+  NSRegularExpression *version_matcher = [NSRegularExpression regularExpressionWithPattern:OS_VERSION_MATCHER options:0 error:nil];
+  NSString *version_info = string_for_key(metadata, @"osVersion");
+  NSTextCheckingResult *match = [[version_matcher matchesInString:version_info options:0 range:NSMakeRange(0, version_info.length)] firstObject];
+  NSString *os_name = match ? [version_info substringWithRange:[match rangeWithName:@"osName"]] : nil;
+  NSString *os_version = match ? [version_info substringWithRange:[match rangeWithName:@"osVersion"]] : nil;
+  NSString *os_kernversion = match ? [version_info substringWithRange:[match rangeWithName:@"buildNumber"]] : nil;
```

Comment:
If one of these `rangeWithName` are not present in the regex grouping this will crash I believe